### PR TITLE
chore(actions): remove warnings related to gcp and docker steps

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

Having warnings that are not helpful in our workflows summaries, makes warning that we should care about less relevant or harder to find.

image

## Solution

- Remove deprecated arguments
- Deactivate Docker summary (not supported by Docker Build Cloud, yet)
  - Use a variable `DOCKER_BUILD_SUMMARY ` and set it to false, as this feature might be available in the future

### Follow-up Work

I also tried solving this warning:
- `Unable to extend GITHUB_TOKEN expiration time due to: GITHUB_TOKEN has expired.`

But it seems we won't be able to do it, based on this discussion: 
- https://github.com/orgs/community/discussions/50472

### PR Author's Checklist

- [X] The PR name will make sense to users.
- [X] The PR provides a CHANGELOG summary.
- [X] The solution is tested.
- [X] The documentation is up to date.
- [X] The PR has a priority label.

### CODE REVIEW

File: .github/workflows/ci-test-librustzcash.yml
Comment (mpguerra) [inline .github/workflows/ci-test-librustzcash.yml:1]: This seems unrelated to the PR title. Should we split this out into another PR?

File: .github/workflows/cd-deploy-nodes-gcp.yml [hunk 1]
```diff
@@ -230,7 +230,6 @@ jobs:
         id: auth
         uses: google-github-actions/auth@v2.1.4
         with:
-          retries: '3'
```
Comment (mpguerra) [inline .github/workflows/cd-deploy-nodes-gcp.yml:233]: Are these the deprecated arguments we are removing? Otherwise I'd like to understand why we are we removing the retries

File: .github/workflows/sub-build-docker-image.yml [hunk 2]
```diff
@@ -75,6 +75,8 @@ jobs:
     permissions:
       contents: 'read'
       id-token: 'write'
+    env:
+      DOCKER_BUILD_SUMMARY: ${{ vars.DOCKER_BUILD_SUMMARY }}
```
Comment (mpguerra) [inline .github/workflows/sub-build-docker-image.yml:79]: In your PR comment you said this is not supported by docker build cloud. Why are we adding it here?

**Review by mpguerra — DISMISSED** [review]
Looks good to me, I just have some questions to try to understand the changes.

File: .github/workflows/cd-deploy-nodes-gcp.yml (hunk 1 above)
Comment (gustavovalverde) [reply .github/workflows/cd-deploy-nodes-gcp.yml:233]: Yes, those have been deprecated 
image

File: .github/workflows/sub-build-docker-image.yml (hunk 2 above)
Comment (gustavovalverde) [reply .github/workflows/sub-build-docker-image.yml:79]: Oh, I'm adding it as a variable, which defaults to false, just in case is supported later on and we'd like to activate it.

File: .github/workflows/ci-test-librustzcash.yml
Comment (gustavovalverde) [reply .github/workflows/ci-test-librustzcash.yml:1]: Oh, it was late and I did not see I added this extra file 😅  . I'll fix this and force-push it.

File: .github/workflows/cd-deploy-nodes-gcp.yml (hunk 1 above)
Comment (gustavovalverde) [reply .github/workflows/cd-deploy-nodes-gcp.yml:233]: I've update the PR comment with this info.

File: .github/workflows/sub-build-docker-image.yml (hunk 2 above)
Comment (gustavovalverde) [reply .github/workflows/sub-build-docker-image.yml:79]: I've update the PR comment with this info.

**Review by upbqdn — APPROVED** [review]

**Comment by gustavovalverde** [comment]
@Mergifyio refresh

**Review by mpguerra — APPROVED** [review]
